deepchem/rl/a3c.py

```diff
@@ -21,10 +21,11 @@ def __init__(self, value_weight, entropy_weight, **kwargs):
     self.entropy_weight = entropy_weight
 
   def create_tensor(self, **kwargs):
-    reward, action, prob, value = [layer.out_tensor for layer in self.in_layers]
-    log_prob = tf.log(prob)
-    policy_loss = -tf.reduce_sum(
-        (reward - value) * tf.reduce_sum(action * log_prob))
+    reward, action, prob, value, advantages = [
+        layer.out_tensor for layer in self.in_layers
+    ]
+    log_prob = tf.log(prob + 0.0001)
+    policy_loss = -tf.reduce_sum(advantages * tf.reduce_sum(action * log_prob))
```

Comment:
I'd also suggest experimenting with different values for `value_weight`.  It's possible the problem is just that your weight is too low, so it's taking a long time to learn the value function.

Also, are you sure it's converged?  100,000 steps isn't very many.